File: src/main/java/org/hwabeag/cashsystem/config/ConfigManager.java
package org.hwabeag.cashsystem.config;

import org.bukkit.configuration.file.FileConfiguration;
import org.hwabeag.cashsystem.CashSystem;

import java.util.HashMap;

public class ConfigManager {
    private static final CashSystem plugin = CashSystem.getPlugin();

    private static final HashMap<String, ConfigMaker> configSet = new HashMap<>();

    public ConfigManager() {
        String path = plugin.getDataFolder().getAbsolutePath();
        configSet.put("cash-system", new ConfigMaker(path, "CashSystem.yml"));
        configSet.put("player", new ConfigMaker(path, "Player.yml"));
        loadSettings();
        saveConfigs();
    }

    public static void reloadConfigs() {
        for (String key : configSet.keySet()){
            plugin.getLogger().info(key);
            configSet.get(key).reloadConfig();
        }
    }

    public static void saveConfigs(){
        for (String key : configSet.keySet())
            configSet.get(key).saveConfig();
    }

    public static FileConfiguration getConfig(String fileName) {
        return configSet.get(fileName).getConfig();
    }

    public static void loadSettings(){
        FileConfiguration StyleSystemConfig = getConfig("cash-system");
        StyleSystemConfig.options().copyDefaults(true);
        StyleSystemConfig.addDefault("cash-system.prefix", "&a&l[CashSystem]&7");
    }
}

File: src/main/kotlin/org/hwabeag/cashsystem/config/ConfigManager.kt
package org.hwabeag.cashsystem.config

import org.bukkit.configuration.file.FileConfiguration
import org.hwabeag.cashsystem.CashSystem

object ConfigManager {
    private lateinit var plugin: CashSystem
    private val configSet: MutableMap<String, ConfigMaker> = linkedMapOf()

    fun initialize(cashSystem: CashSystem) {
        plugin = cashSystem
        val path = plugin.dataFolder.absolutePath

        configSet["cash-system"] = ConfigMaker(path, "CashSystem.yml")
        configSet["setting"] = ConfigMaker(path, "config.yml")
        configSet["player"] = ConfigMaker(path, "Player.yml")

        loadDefaults()
        saveConfigs()
    }

    fun reloadConfigs() {
        configSet.values.forEach { it.reloadConfig() }
    }

    fun saveConfigs() {
        configSet.values.forEach { it.saveConfig() }
    }

    fun getConfig(fileName: String): FileConfiguration {
        return requireNotNull(configSet[fileName]) { "설정 파일 키를 찾을 수 없습니다: $fileName" }.getConfig()
    }

    private fun loadDefaults() {
        val cashSystemConfig = getConfig("cash-system")
        cashSystemConfig.options().copyDefaults(true)
        cashSystemConfig.addDefault("cash-system.prefix", "&a&l[CashSystem]&7")

        val settingConfig = getConfig("setting")
        settingConfig.options().copyDefaults(true)
        settingConfig.addDefault("database.type", "sqlite")
        settingConfig.addDefault("database.sqlite.file", "cashsystem.db")
        settingConfig.addDefault("database.mysql.host", "127.0.0.1")
        settingConfig.addDefault("database.mysql.port", 3306)
        settingConfig.addDefault("database.mysql.database", "cashsystem")
        settingConfig.addDefault("database.mysql.username", "root")
        settingConfig.addDefault("database.mysql.password", "change-me")
        settingConfig.addDefault("database.mysql.use-ssl", false)
        settingConfig.addDefault("database.migrate-player-yml", true)
        settingConfig.addDefault("shop-item-lore", listOf("&f&l- 현재 구매가 %amount%원", "&f&l- 클릭시 구매됩니다."))
    }
}

